src/library/scala/collection/immutable/RedBlackTree.scala

```diff
@@ -546,14 +613,33 @@ private[collection] object NewRedBlackTree {
                            @(inline @getter @setter)     private var _right: Tree[A, _],
                            @(inline @getter @setter)     private var _count: Int)
   {
-    @`inline` private[NewRedBlackTree] def isMutable: Boolean = (_count & colourMask) == 0
+    //used in testing
+    @`inline` private[immutable] def isMutable: Boolean = (_count & colourMask) == 0
     // read only APIs
     @`inline` private[NewRedBlackTree] final def count = {
       //devTimeAssert((_count & 0x7FFFFFFF) != 0)
       _count & colourMask
     }
     //retain the colour, and mark as mutable
     @`inline` private def mutableRetainingColour = _count & colourBit
+    def debugToString: String = {
+      val b = new StringBuilder
+      def loop(parent: Tree[A, B], tree: Tree[A, B], depth: Int): Unit = {
+        b ++= ("  " * depth)
+        if (tree == null) b ++= "Empty\n"
+        else {
+          b ++= s"${if (tree.isRed) "RedTree" else "BlackTree"}(${tree.key}, ${tree.value})"
```

Comment:
Could append "#${System.identityHashCode(this)}` to make it easier to eyeball structural sharing.